plugins/com.google.cloud.tools.eclipse.login/src/com/google/cloud/tools/eclipse/login/GoogleLoginService.java

```diff
@@ -100,13 +100,11 @@ public GoogleLoginService() {}
   @VisibleForTesting
   GoogleLoginService(OAuthDataStore dataStore, LoginServiceUi uiFacade, LoggerFacade loggerFacade) {
     this(new GoogleLoginState(Constants.getOAuthClientId(), Constants.getOAuthClientSecret(),
-                              OAUTH_SCOPES, dataStore, uiFacade, loggerFacade),
-         dataStore, uiFacade, loggerFacade);
+                              OAUTH_SCOPES, dataStore, uiFacade, loggerFacade), uiFacade);
   }
 
   @VisibleForTesting
-  GoogleLoginService(GoogleLoginState loginState,
-      OAuthDataStore dataStore, LoginServiceUi uiFacade, LoggerFacade loggerFacade) {
+  GoogleLoginService(GoogleLoginState loginState, LoginServiceUi uiFacade) {
```

Comment:
Looks like we can also remove `uiFacade`, which in turn allows us to remove the field `loginServiceUi` from this class as well.